contracts/registries/YRegistry.sol

```diff
@@ -0,0 +1,237 @@
+      token = IVault(_vault).token(); // Use non-wrapped vault
+    } else {
+      token = IVault(vault).token();
+    }
+    
+    if (isDelegated) {
+      strategy = IController(controller).strategies(vault);
+    } else {
+      strategy = IController(controller).strategies(token);
+    }
+    
+    // Check if vault is set on controller for token
+    address controllerVault = address(0);
+    if (isDelegated) {
+      controllerVault = IController(controller).vaults(strategy);
+    } else {
+      controllerVault = IController(controller).vaults(token);
+    }
+    require(controllerVault == vault, "Controller vault address does not match"); // Might happen on Proxy Vaults
+    
+    // Check if strategy has the same token as vault
+    if (isWrapped) {
+      address underlying = IVault(vault).underlying();
+      require(underlying == token, "WrappedVault token address does not match"); // Might happen?
+    } else if (isDelegated) {
+
+    } else {
+      address strategyToken = IStrategy(strategy).want();
+      require(token == strategyToken, "Strategy token address does not match"); // Might happen?
+    }
+
+    return (controller, token, strategy, isWrapped, isDelegated);
+  }
+
+  // Vaults getters
+  function getVault(uint index) external view returns (address vault) {
+    return vaults.at(index);
+  }
+
+  function getVaultsLength() external view returns (uint) {
+    return vaults.length();
+  }
+
+  function getVaults() external view returns (address[] memory) {
+    address[] memory vaultsArray = new address[](vaults.length());
+    for (uint i = 0; i < vaults.length(); i++) {
+      vaultsArray[i] = vaults.at(i);
+    }
+    return vaultsArray;
+  }
+
+  function getVaultInfo(address _vault) external view returns (
+    address controller,
+    address token,
+    address strategy,
+    bool isWrapped,
+    bool isDelegated
+  ) {
+    (controller, token, strategy, isWrapped, isDelegated) = getVaultData(_vault);
+    return (
+      controller,
+      token,
+      strategy,
+      isWrapped,
+      isDelegated
+    );
+  }
+
+  function getVaultsInfo() external view returns (
+    address[] memory controllerArray,
+    address[] memory tokenArray,
+    address[] memory strategyArray,
+    bool[] memory isWrappedArray,
+    bool[] memory isDelegatedArray
+  ) {
+    controllerArray = new address[](vaults.length());
+    tokenArray = new address[](vaults.length());
+    strategyArray = new address[](vaults.length());
+    isWrappedArray = new bool[](vaults.length());
+    isDelegatedArray = new bool[](vaults.length());
+
+    for (uint i = 0; i < vaults.length(); i++) {
+      (address _controller, address _token, address _strategy, bool _isWrapped, bool _isDelegated) = getVaultData(vaults.at(i));
+      controllerArray[i] = _controller;
+      tokenArray[i] = _token;
+      strategyArray[i] = _strategy;
+      isWrappedArray[i] = _isWrapped;
+      isDelegatedArray[i] = _isDelegated;
```

Comment:
What happens if one of the lengths is shorter than another? Can we have a `require` check of all matching?